engines/alg/alg.cpp

```diff
@@ -0,0 +1,105 @@
+/* ScummVM - Graphic Adventure Engine
+ *
+ * ScummVM is the legal property of its developers, whose names
+ * are too numerous to list here. Please refer to the COPYRIGHT
+ * file distributed with this source distribution.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "common/config-manager.h"
+#include "engines/util.h"
+
+#include "alg/alg.h"
+#include "alg/detection.h"
+#include "alg/game.h"
+#include "alg/logic/game_bountyhunter.h"
+#include "alg/logic/game_crimepatrol.h"
+#include "alg/logic/game_drugwars.h"
+#include "alg/logic/game_johnnyrock.h"
+#include "alg/logic/game_maddog.h"
+#include "alg/logic/game_maddog2.h"
+#include "alg/logic/game_spacepirates.h"
+
+namespace Alg {
+
+AlgEngine::AlgEngine(OSystem *syst, const AlgGameDescription *gd)
+	: Engine(syst), _gameDescription(gd) {
+	switch (gd->gameType) {
+	case Alg::GType_CRIME_PATROL: {
+		GameCrimePatrol *game = new GameCrimePatrol(this, gd);
+		_debugger = new DebuggerCrimePatrol(game);
+		_game = game;
+		break;
+	}
+	case Alg::GType_DRUG_WARS: {
+		GameDrugWars *game = new GameDrugWars(this, gd);
+		_debugger = new DebuggerDrugWars(game);
+		_game = game;
+		break;
+	}
```

Comment:
```suggestion
	case Alg::GType_CRIME_PATROL: {
		_game = new GameCrimePatrol(this, gd);
		_debugger = new DebuggerCrimePatrol(_game);
		break;
	}
	case Alg::GType_DRUG_WARS: {
		_game = new GameDrugWars(this, gd);
		_debugger = new DebuggerDrugWars(_game);
		break;
	}
```
Same for the cases below

Reply:
If i implement that change it does not compile anymore:

engines/alg/alg.cpp: In constructor 'Alg::AlgEngine::AlgEngine(OSystem*, const Alg::AlgGameDescription*)':
engines/alg/alg.cpp:43:53: error: invalid conversion from 'Alg::Game*' to 'Alg::GameCrimePatrol*' [-fpermissive]
   43 |                 _debugger = new DebuggerCrimePatrol(_game);